src/operations/execute_operation.ts

```diff
@@ -210,6 +205,13 @@ async function tryOperation<T extends AbstractOperation, TResult = ResultTypeFro
     signal: operation.options.signal
   });
 
+  if (
+    session?.snapshotEnabled &&
+    maxWireVersion(topology) < MIN_SUPPORTED_SNAPSHOT_READS_WIRE_VERSION
+  ) {
+    throw new MongoCompatibilityError('Snapshot reads require MongoDB 5.0 or later');
+  }
```

Comment:
When this check was done by ServerCapabilities it was written as

```js
this.maxWireVersion >= 13;
```

Which is the opposite right? 🤔